Chained.h

```diff
@@ -2,17 +2,206 @@
+        ++numPairs;
+        if((static_cast<float>(numPairs))/(static_cast<float>(hashmapSize)) > MAX_LOAD_FACTOR){
+            doubleSize();
+        }
+    }
+
+    void remove(Key k) {
+        uint64_t index = hash(k);
+        for(auto i = (*hashmap)[index].begin(); i != (*hashmap)[index].end(); ++i) {
+            if(k == i->key){
+                (*hashmap)[index].erase(i);
+                --numPairs;
+                return;
+            }
+        }
+    }
+
+    void display(bool print){
+        std::vector<uint64_t> elements;
+        std::vector<uint64_t> collisions;
+        uint64_t numWith0 = 0;
+        uint64_t numWith1 = 0;
+        uint64_t numWithMore = 0;
+
+        //loop through and print the data structure while storing important information
+        for(uint64_t i = 0; i < hashmapSize; i++){
+            if(print) {
+                std::cout << i << ": |" << (*hashmap)[i].size() << "| ";
+            }
+            elements.push_back(static_cast<uint64_t>((*hashmap)[i].size()));
+            if((*hashmap)[i].empty()) {
+                ++numWith0;
+                collisions.push_back(0);
+            } else {
+                collisions.push_back(static_cast<uint64_t>((*hashmap)[i].size()-1));
+                if((*hashmap)[i].size() == 1){
+                    ++numWith1;
+                }
+                else{
+                    ++numWithMore;
+                }
+            }
+            for(auto kvp : (*hashmap)[i]){
+                if(print) {
+                    printKeyValuePair(kvp);
+                }
+            }
+            if(print) {
+                std::cout << "\n";
+            }
+        }
+        std::cout << std::endl;
+
+        //stats output
+        if(elements.empty()){
+            std::cout << "No Elements\n";
+        } else {
+            std::sort(elements.begin(), elements.end());
+            Statistic* elementStats = evaluateStats(&elements);
+            std::cout << "Total Elements: " << elementStats->total << "\n";
+            std::cout << "Mean Elements: " << elementStats->mean << "\n";
+            std::cout << "Stdev Elements: " << elementStats->stdev << "\n";
+
+            std::sort(collisions.begin(), collisions.end());
+            Statistic* collisionStats = evaluateStats(&collisions);
+            std::cout << "Total Collisions: " << collisionStats->total << "\n";
+            std::cout << "Mean Collisions: " << collisionStats->mean << "\n";
+            std::cout << "Stdev Collisions: " << collisionStats->stdev << "\n";
+
+            std::cout << "Total Number of Lists: " << hashmapSize << "\n";
+            std::cout << "Number of Lists With Zero Elements: " << numWith0 << "\n";
+            std::cout << "Number of Lists with One Element: " << numWith1 << "\n";
+            std::cout << "Number of Lists with more than One Element: " << numWithMore << "\n";
+            //I could make a display of all element sizes. Box plot? Scatterplot?
```

Comment:
I think histogram would be the right call.
But I'd settle for mean size of collision list.